assets/js/carouselClass.js

```diff
@@ -0,0 +1,105 @@
+export class CarouselClass {
+  constructor(container) {
+    // Created variables
+    this.container = container;
+    this.carouselList = this.container.querySelector(".carousel-list"),
+    this.carouselItems = this.container.querySelectorAll(".carousel-list__item"),
+    this.carouselWrap = this.container.querySelector(".carousel-wrap"),
+    this.btnNext = this.container.querySelector(".carousel-next"),
+    this.btnPrev = this.container.querySelector(".carousel-prev"),
+    this.pagination = this.container.querySelectorAll(".pagination-bullet");
+
+    // Calculated variables
+    this.carouselWrapWidth = this.carouselWrap.offsetWidth,
+    this.carouselItemsLength = this.carouselItems.length,
+    this.carouselItemsActiveIndex = 0,
+    this.isResizing = false;
+
+
+  }
+
+  // Add or disable switch transition
+  addTransition() {
+    return (this.carouselList.style.transition = this.isResizing
+      ? ""
+      : "transform 0.5s ease-in-out");
+  };
+
+  // Set active class to carousel item
+  addActiveToItem() {
+    this.carouselItems.forEach((item, index) => {
+      return index ===  this.carouselItemsActiveIndex
+        ? item.classList.add("carousel-list__item--active")
+        : item.classList.remove("carousel-list__item--active");
+    });
+  };
+
+  // Add/remove class to pagination bullet
+  setActivePagination() {
+    this.pagination.forEach((item, index) => {
+      return index === this.carouselItemsActiveIndex
+        ? item.classList.add("pagination-bullet--active")
+        : item.classList.remove("pagination-bullet--active");
+    });
+  };
+
+  // Navigate carousel on pagination click
+  moveCarouselOnPagination() {
+    this.pagination.forEach((item, index) => {
+      item.addEventListener("click", () => {
+        this.carouselItemsActiveIndex = index;
+        this.moveCarousel();
+      });
+    });
+  };
+
+  // On window resize change carouselWrapWidth to atjust transform
+  changeSize() {
```

Comment:
Consider naming callbacks with `on` prefix.

```suggestion
  onWindowResize() {
```